src/scripts/slack/message.go

```diff
@@ -0,0 +1,37 @@
+package slack
+
+import (
+	"fmt"
+	"os"
+
+	"github.com/CircleCI-Public/slack-orb-go/src/scripts/httputils"
+	"github.com/CircleCI-Public/slack-orb-go/src/scripts/jsonutils"
+)
+
+type Message struct {
+	Template     string
+	AccessToken  string
+	IgnoreErrors bool
+	Channels     []string
+}
+
+func (s *Message) PostMessage(channel string) {
+	jsonWithChannel, _ := jsonutils.ApplyFunctionToJSON(s.Template, jsonutils.AddRootProperty("channel", channel))
+	fmt.Printf("Posting the following JSON to Slack:\n%s\n", jsonWithChannel)
+
+	headers := map[string]string{
+		"Content-Type":  "application/json",
+		"Authorization": "Bearer " + s.AccessToken,
+	}
+	response, statusCode, _ := httputils.SendHTTPRequest("POST", "https://slack.com/api/chat.postMessage", jsonWithChannel, headers)
```

Comment:
one of the related problems with not returning a result (or an error) from PostMessage is highlighted here - rather than using the potentially non-nil error being returned from SendHTTPRequest, it is being ignored. Ideally any underlying error coming from the POST (i.e. something that's not a known failure case from the Slack API) is unrepresentable by this code path - if the network connection flakes or there's a transient DNS issue, it will be silently ignored until the response body (that either won't exist, or will be in a different shape than you expect) is introspected.

That said, some sort of retry mechanism for non-success error codes (e.g. HTTP 500s) should likely be put in as well (networks are flaky and unreliable).

Reply:
Shameless plug for the [ex/httpclient package](https://pkg.go.dev/github.com/circleci/ex@v1.0.8178-30b1ec6/httpclient) (it's a public repo in the private org, no issues using it from this orb)